src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/teleporter/PortableTeleporter.java

```diff
@@ -0,0 +1,54 @@
+package io.github.thebusybiscuit.slimefun4.implementation.items.teleporter;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+import org.bukkit.block.BlockFace;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.ItemStack;
+
+import io.github.thebusybiscuit.slimefun4.core.attributes.Rechargeable;
+import io.github.thebusybiscuit.slimefun4.core.handlers.ItemUseHandler;
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import io.github.thebusybiscuit.slimefun4.implementation.items.SimpleSlimefunItem;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+
+/**
+ * This item allows a {@link Player} to access and teleport to his waypoints
+ * from anywhere.
+ *
+ * @author martinbrom
+ *
+ * @see Teleporter
+ */
+public class PortableTeleporter extends SimpleSlimefunItem<ItemUseHandler> implements Rechargeable {
+
+    private static final float COST = 2F;
```

Comment:
I added a new `FloatRangeSetting` but there are two more options with their potential drawbacks:
- Use `DoubleRangeSetting` instead but we could get overflow if the max value is specified wrongly
- Use some sort of generic numerical ItemSetting but that would maybe introduce errors in precision when comparing the boxed types and also would get rid of the implicit null-safety because there are no primitive generics yet